crates/noirc_evaluator/src/ssa_refactor/opt/inlining.rs

```diff
@@ -345,43 +346,16 @@ impl<'function> PerFunctionContext<'function> {
         let arguments = vecmap(arguments, |arg| self.translate_value(*arg));
         let new_results = self.context.inline_function(ssa, function, &arguments);
         let new_results = InsertInstructionResult::Results(&new_results);
-        Self::insert_new_instruction_results(&mut self.values, old_results, new_results);
+        self.values.insert_new_instruction_results(old_results, new_results);
     }
 
     /// Push the given instruction from the source_function into the current block of the
     /// function being inlined into.
     fn push_instruction(&mut self, id: InstructionId) {
-        let instruction = self.source_function.dfg[id].map_values(|id| self.translate_value(id));
-        let results = self.source_function.dfg.instruction_results(id);
-
-        let ctrl_typevars = instruction
-            .requires_ctrl_typevars()
-            .then(|| vecmap(results, |result| self.source_function.dfg.type_of_value(*result)));
-
+        let (instruction, results, ctrl_typevars) =
+            self.values.map_instruction(&self.source_function.dfg, id);
```

Comment:
I find this a bit confusing to read - it seems unintuitive that `map_instruction` should return unmapped results.

Reply:
Perhaps `dfg` just needs an accessor for an instruction's `ctrl_typevars`. It wouldn't save as many lines of code, but I think it would be easier to follow than having to remember `map_instruction`'s return behaviour.